nickgerace/gfold
# Enable multi-platform development

TLDR: remove barriers to multi-platform development, like Bash and Make.
Moreover, remove the nightly fmt dependency and improve testing to catch
potential corner cases on multiple platforms.

Core:
- Rename "Reports" to "LabeledReports" since "Reports" is not of implied
  type Vec<Report>

Testing:
- Add second half to the integration test: comparing found reports with
  expected reports
  - Essentially, we run the core gfold loop _again_, but only generate
    reports and skip displaying them to stdout

CI:
- Removed nightly fmt check and job
- Moved fmt check to stable
- Update bors toml to match changes

Scripts:
- Convert scripts from Bash scripts to independent crates
  (multi-platform friendly)
- Add README to scripts directory
- Ensure script crates do not have their Cargo lockfile tracked in Git

Docs:
- Adding DEVELOPING file to replace Makefile
- Move THANKS and RELEASE files to new docs directory, along with
  DEVELOPING
- Replace make commands in README and RELEASE files

Misc:
- Remove Rust nightly dependency since it was solely for imports
  granulaity for rustfmt
  - This repository is once again only reliant on stable Rust
    toolchains
- Remove "rust-version" from Cargo toml since it was unused
- Remove Makefile entirely in favor of DEVELOPING file (multi-platform
  friendly)
- Update markdown code block languages as "shell" instead of "bash" to
  be multi-platform friendly
